### ISSUE TITLE
target-size rule should flag tabindex=-1 widgets for review

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#3695]
Elements with tabindex="-1" are tricky in the target-size rule. Patterns like tab bars use tabindex="-1" on inactive tabs. These are still click targets, but to get to them with keyboard, arrow navigation is used instead of tab. On the other hand there are examples where tabindex="-1" is used for disabled / inactive components.

It's impossible to definitively know if tabindex="-1" means a component won't respond to a click event. I think the only safe thing to to is to flag them for review. The only exception I think is if the control with tabindex="-1" is a descendant of some other focusable control. This was tackled in https://github.com/dequelabs/axe-core/pull/3682

**Comment by padmavemulapati** [comment]
Validated with the latest develop branch code base, 
For the test script:
```
<a role="link" aria-label="play" tabindex="0" style="display:inline-block" id="target">
    <button style="margin:1px; line-height:20px">Play</button>
  </a>

  <a role="link" aria-label="play" tabindex="0" style="display:inline-block" id="target">
    <button tabindex="-1" style="margin:1px; line-height:20px">Play</button>
  </a>

  <p>
    <a
      role="link"
      aria-label="play"
      tabindex="0"
      style="display: inline-block"
      id="pass4"
    >
      <button
        style="margin: 1px; line-height: 20px"
        id="inapplicable7"
        tabindex="-1"
      >
        Play
      </button>
    </a>
  </p>
  
  <p>
    <a role="link" tabindex="0" id="pass5">
      Wide link <button id="fail7">y</button>
    </a>
  </p>
  ```
seeing `target-size` failure on `<a role="link" aria-label="play" tabindex="0" style="display:inline-block" id="target">
    <button style="margin:1px; line-height:20px">Play</button>
  </a>` 
and 
`<button id="fail7">y</button>`


![Image](https://user-images.githubusercontent.com/47409579/196843258-6565cfd8-120f-4d72-aeb4-768118ad5576.png)


also it is passing on 



![Image](https://user-images.githubusercontent.com/47409579/196843300-60389f69-2004-4d3d-b77a-36b804638872.png)
